aeon/io/device.py

```diff
@@ -1,41 +1,40 @@
 import inspect
 
 
-def compositeStream(pattern, *args):
-    """Merges multiple data streams into a single composite stream."""
-    composite = {}
+def register(pattern, *args):
+    """Merges multiple Readers into a single registry."""
+    registry = {}
     if args:
-        for stream in args:
-            if inspect.isclass(stream):
-                for method in vars(stream).values():
+        for binder_fn in args:
+            if inspect.isclass(binder_fn):
+                for method in vars(binder_fn).values():
                     if isinstance(method, staticmethod):
-                        composite.update(method.__func__(pattern))
+                        registry.update(method.__func__(pattern))
             else:
-                composite.update(stream(pattern))
-    return composite
+                registry.update(binder_fn(pattern))
+    return registry
 
 
 class Device:
-    """Groups multiple data streams into a logical device.
+    """Groups multiple Readers into a logical device.
 
-    If a device contains a single stream with the same pattern as the device
-    `name`, it will be considered a singleton, and the stream reader will be
-    paired directly with the device without nesting.
+    If a device contains a single stream reader with the same pattern as the device `name`, it will be 
+    considered a singleton, and the stream reader will be paired directly with the device without nesting.
 
     Attributes:
         name (str): Name of the device.
-        args (Any): Data streams collected from the device.
+        args (any): A binder function or class that returns a dictionary of Readers.
```

Comment:
`Any` here meant the [Any type](https://docs.python.org/3/library/typing.html#the-any-type) so it should either be left with its original casing, or replaced with something else from the typing module.

A possible alternative to "binder function" could be "stream selector" or "stream accessor", since basically what the function brings in addition to the reader is the pattern to find for stream data files.

Another option is simply to call this concept a "stream". We can discuss better on a future meeting.